modules/codegen/src/smithy4s/codegen/internals/Renderer.scala

```diff
@@ -560,20 +576,69 @@ private[internals] class Renderer(compilationUnit: CompilationUnit) { self =>
         block(
           line"def liftError(throwable: Throwable): $option[$errorName] = throwable match"
         ) {
-          op.errors.collect { case Type.Ref(pkg, name) =>
-            line"case e: ${NameRef(pkg + "." + name)} => $some($errorName.${name}Case(e))"
-          } ++ List(line"case _ => $none")
+          if (scala3Unions) {
+            List(
+              line"case e: $errorName => $some(e)",
+              line"case _ => $none"
+            )
+          } else {
+            op.errors.collect { case Type.Ref(pkg, name) =>
+              line"case e: ${NameRef(pkg + "." + name)} => $some($errorName.${name}Case(e))"
+            } ++ List(line"case _ => $none")
+          }
         },
-        block(
-          line"def unliftError(e: $errorName): Throwable = e match"
-        ) {
-          op.errors.collect { case Type.Ref(_, name) =>
-            line"case $errorName.${name}Case(e) => e"
+        if (scala3Unions) line"def unliftError(e: $errorName): Throwable = e"
+        else
+          block(
+            line"def unliftError(e: $errorName): Throwable = e match"
+          ) {
+            op.errors.collect { case Type.Ref(_, name) =>
+              line"case $errorName.${name}Case(e) => e"
+            }
           }
-        }
       )
   }
 
+  private def renderErrorUnion(
```

Comment:
let's rename it to `renderErrorAsScala3Union` to avoid confusing potential readers, as "unions" can refer to smithy4 unions.